pkg/utils/image.go

```diff
@@ -0,0 +1,56 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// WARNING: not supporting CIG images yet. Awaiting AKS machine API support.
+func GetAKSMachineNodeImageVersionFromImageID(imageID string) (string, error) {
+	if strings.HasPrefix(imageID, "/CommunityGalleries") {
+		// Awaiting AKS machine API support
+		return "", fmt.Errorf("CIG images are not supported yet for AKS machines, consider not using PROVISION_MODE=aksmachineapi: %s", imageID)
+	} else {
+		return GetAKSMachineNodeImageVersionFromSIGImageID(imageID)
+	}
+}
```

Comment:
ideally we would have some image parsing helpers for the different places we parse and manage images, similar to the VnetSubnetIDComponents helper